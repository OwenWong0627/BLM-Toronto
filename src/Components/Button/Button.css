:root {
  --primary: #fff;
}

.btn--primary {
  background-color: var(--primary);
  color: #242424;
  border: 1px solid var(--primary);
  cursor: pointer;
  outline: none;
}

.btn--outline {
  background-color: #00BFFF;
  color: #fff;
  border: 1px solid var(--primary);
  border-radius: 7px;
  transition: all 0.3s ease-out;
  cursor: pointer;
  outline: none;
}

.btn--medium {
  padding: 8px 15px;
  font-size: 18px;
}

.btn--large {
  padding: 12px 26px;
  font-size: 20px;
}

.btn--large:hover,
.btn--medium:hover {
  transition: all 0.3s ease-out;
  background: #50C878;
  color: #000;
  transition: 250ms;
}
.btn--large:hover,
.btn--medium:hover,
.btn--large:focus,
.btn--medium:focus {
  box-shadow: inset 0 0 0 2em var(--hover);
}