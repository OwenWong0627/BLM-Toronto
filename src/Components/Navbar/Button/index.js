import React from 'react';
import './index.css';
import { BrowserRouter as Router, Link } from 'react-router-dom';

const STYLES = ['btn--primary', 'btn--outline'];

const SIZES = ['btn--medium', 'btn--large'];

export const Button = ({
    children,
    type,
    onClick,
    buttonStyle,
    buttonSize
}) => {
    const checkButtonStyle = STYLES.includes(buttonStyle) ? buttonStyle : STYLES[0];

    const checkButtonSize = SIZES.includes(buttonSize) ? buttonSize : SIZES[0];

    return (
        <Router>
            <Link to='/location-search' className='btn-mobile'>
                <button
                    className={`btn ${checkButtonStyle} ${checkButtonSize}`}
                    onClick={onClick}
                    type={type}
                >
                    {children}
                </button>
            </Link>
        </Router>
    )

};

export default Button;
