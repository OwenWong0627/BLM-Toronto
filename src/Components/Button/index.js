import React from 'react';
import './Button.css';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';

const STYLES = ['btn--primary', 'btn--outline'];

const SIZES = ['btn--medium', 'btn--large'];

/**
 * Component for the button in the navbar(it will not appear in small screen widths).
 * The component will take in two required props: label for the text in the button and type for the type of the button tag
 */
export const Button = ({
   label,
   type,
   onClick,
   buttonStyle,
   buttonSize
}) => {
   const checkButtonStyle = STYLES.includes(buttonStyle) ? buttonStyle : STYLES[0];

   const checkButtonSize = SIZES.includes(buttonSize) ? buttonSize : SIZES[0];

   return (
      <Link to='/find-business' data-testid="button-link">
         <button
            data-testid="button"
            className={`btn ${checkButtonStyle} ${checkButtonSize}`}
            onClick={onClick}
            type={type}
         >
            {label}
         </button>
      </Link>
   )
};

/**
 * propType Documentation
 */
Button.propTypes = {
   label: PropTypes.string.isRequired,
   type: PropTypes.string.isRequired,
   onClick: PropTypes.func,
   buttonStyle: PropTypes.string,
   buttonSize: PropTypes.string,
}

export default Button;
