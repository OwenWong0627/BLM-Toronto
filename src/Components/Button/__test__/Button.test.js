import React from 'react';
import ReactDOM from 'react-dom';
import Button from '..';

import { BrowserRouter as Router } from 'react-router-dom';
import { render, cleanup } from '@testing-library/react';
import '@testing-library/jest-dom';
import renderer from 'react-test-renderer';

afterEach(cleanup);
it('Renders without crashing', () => {
   const div = document.createElement("div");
   ReactDOM.render(<Router><Button label="haha" type="button" /></Router>, div);
});

it('Renders button correctly', () => {
   const { getByTestId } = render(<Router><Button label="Click to Find Business" buttonStyle="btn--primary" type="button" /></Router>);
   expect(getByTestId('button')).toHaveTextContent("Click to Find Business");
   expect(getByTestId('button')).toHaveClass("btn--primary");
   expect(getByTestId('button')).toHaveAttribute("type", "button");
});

it('Renders button correctly', () => {
   const { getByTestId } = render(<Router><Button label="CLICK ME PLEASE" buttonStyle="btn--outline" buttonSize="btn--medium" type="button" /></Router>);
   expect(getByTestId('button')).toHaveTextContent("CLICK ME PLEASE");
   expect(getByTestId('button')).toHaveClass("btn--outline");
   expect(getByTestId('button')).toHaveClass("btn--medium");
   expect(getByTestId('button')).not.toHaveAttribute("type", "submit");
});

it('Renders button with correct href', () => {
   const { getByTestId } = render(<Router><Button label="Find Business" buttonStyle="btn--outline" type="button" /></Router>);
   expect(getByTestId('button-link')).toHaveAttribute("href", "/find-business");
});

it('matches snapshot', () => {
   const tree = renderer.create(<Router><Button label="CLICK ME PLEASE" buttonStyle="btn--outline" type="button" /></Router>).toJSON();
   expect(tree).toMatchSnapshot();
});