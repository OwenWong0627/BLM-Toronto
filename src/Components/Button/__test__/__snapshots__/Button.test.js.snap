// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<a
  data-testid="button-link"
  href="/find-business"
  onClick={[Function]}
>
  <button
    className="btn btn--outline btn--medium"
    data-testid="button"
    type="button"
  >
    CLICK ME PLEASE
  </button>
</a>
`;
